modules/oke/cluster.tf

```diff
@@ -71,11 +71,16 @@ resource "oci_containerengine_cluster" "k8s_cluster" {
       defined_tags  = lookup(var.defined_tags,"service_lb",{})
     }
 
-    service_lb_subnet_ids = var.preferred_load_balancer == "public" ? [var.cluster_subnets["pub_lb"]] : [var.cluster_subnets["int_lb"]]
+    service_lb_subnet_ids = [var.cluster_subnets[local.lb_subnet]]
   }
 
   lifecycle {
     ignore_changes = [defined_tags, freeform_tags, cluster_pod_network_options]
+
+    precondition {
+      condition     = var.cluster_subnets[local.lb_subnet] != ""
+      error_message = "preferred_load_balancer references unexisting load balancer."
```

Comment:
Please provide a more intuitive message instead.

Reply:
What do u thing about
> Preferred load balancer references unexisting load balancer. Please check variables preferred_load_balancer and load_balancers.

Reply:
You should guide them to what is wrong in their configuration e.g. ```preferred_load_balancer``` cannot be an empty string. Otherwise, it's quite meaningless as there's no load balancer created with this module.

Reply:
`preferred_load_balancer` is not an empty string.

User may archive such configuration if he specify `load_balancers = "internal"` and doesn't specify preferred_load_balancer -- such configuration creates only internal load balancer, but `preffered_load_balancer` is pointing to the default value `public`. The problem is (and I've experienced it when tried module at first) that you may not be aware o the `preferred_load_balancer` variable at all.

So the oke module fails to find the public lb subnet.
I've tried to point a user to the variables `preferred_load_balancer` and `load_balancers` as it will help him to find what is triggering an error.